gateway/handler_error.go

```diff
@@ -311,6 +314,29 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			log.WithError(err).Error("could not store analytic record")
 		}
 	}
+
+	// Print the transaction logs for error situations if enabled. Success transaction
+	// logs will be handled by the "handler_success.go"
+	if e.Spec.GlobalConfig.AccessLogs.Enabled {
+		// Don't print the full token, handle as obfuscated key or hashed key for security reasons
+		hashKeys := e.Gw.GetConfig().HashKeys
+		key := ""
+
+		if !hashKeys {
+			key = e.Gw.obfuscateKey(token)
+		} else {
+			key = storage.HashKey(token, hashKeys)
+		}
+
+		accessLog := httputil.NewAccessLogRecord(e.Spec.APIID, key, e.Spec.OrgID)
+		accessLog.WithLatency(&analytics.Latency{})
+		accessLog.WithRequest(r)
+		accessLog.WithResponse(response)
+
+		logger := accessLog.Logger(log)
```

Comment:
It would be better to return logrus.Fields from this;

Expected: `log.WithFields(accessLog.Fields()).Info()`

- Rename `accessLog.Logger(*logrus.Logger) *logrus.Entry` into `Fields() logrus.Fields`